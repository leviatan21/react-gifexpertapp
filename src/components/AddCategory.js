import React, { useState } from 'react';
import PropTypes from 'prop-types'
//	rafcp
export const AddCategory = ( { setCategories } ) => {
	//
	const initialState = "";
	const [ inputValue, setInputValue ] = useState( initialState );
	//
	const handleInputChange = ( e ) => {
		//	console.log( 'handleInputChange', e.target.value );
		setInputValue( e.target.value );
	}
	//
	const handleSubmit = ( e ) => {
		e.preventDefault( );
		//	console.log( 'handleSubmit', inputValue );
		if ( inputValue.trim().length > 2 ) {
			setCategories( state => [ inputValue, ...state ] );
			setInputValue( initialState );
		}
	}
	//
	return (
		<form onSubmit={ handleSubmit }>
			<input 
				type="text"
				value={ inputValue }
				onChange={ handleInputChange }
				placeholder="..."
			/>
		</form>
	)
}
//
AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired
}