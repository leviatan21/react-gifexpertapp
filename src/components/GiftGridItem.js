import React from 'react';
import PropTypes from 'prop-types'
//
export const GiftGridItem = ( { title, url } ) => {
	//
	return (
		<div className="card animate__animated animate__fadeIn animate__delay-2s">
			<img src={ url } alt={ title } />
			<p>{ title }</p>
		</div>
	)
}
//
GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}