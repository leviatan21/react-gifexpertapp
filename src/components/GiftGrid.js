import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GiftGridItem } from './GiftGridItem';
//
export const GiftGrid = ( { category } ) => {
	//
	const { data:images, loading } = useFetchGifs( category );
	//
	return (
		<>
			<h3 className="animate__animated animate__fadeIn">{ category }</h3>
			{ loading && <p className="loading animate__animated animate__flash">Cargando</p> }
			<div className="card-grid">
				{ images.map( (gif) => <GiftGridItem key={ gif.id } { ...gif } /> ) }
			</div>
		</>
	)
}
//
GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
}
