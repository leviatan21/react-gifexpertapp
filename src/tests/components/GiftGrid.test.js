
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GiftGrid } from '../../components/GiftGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock( '../../hooks/useFetchGifs' );
//
describe( 'Puebas en <GiftGrid />', () => {
	//
	const gifs = [
		{
			id:		'3ohc1292yKn6Z1saGs',
			title:	'getpunches GIF by mannyjammy',
			url:	'https://media3.giphy.com/media/3ohc1292yKn6Z1saGs/giphy.gif?cid=b1bbf17c1qvapj1ew4s6nuveegu7o50xrwvlagvr77c1reso&rid=giphy.gif'
		},
		{
			id:		'l41m1H2BYnLRV26QM',
			title:	'Bruce Lee One Inch Punch GIF by Supercompressor',
			url:	'https://media2.giphy.com/media/l41m1H2BYnLRV26QM/giphy.gif?cid=b1bbf17c1qvapj1ew4s6nuveegu7o50xrwvlagvr77c1reso&rid=giphy.gif'
		}
	];
	const category	= 'Goku';
	//
	test( 'Componente GiftGrid', ( ) => {
		//
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true
		});
		//
		const wrapper		= shallow( <GiftGrid category={ category } /> );
		const findHeader3	= wrapper.find( 'h3' );
		const findLoading	= wrapper.find( '.loading' );
		const findCard		= wrapper.find( '.card-grid' );
		const findGridItem	= wrapper.find( 'GiftGridItem' );
		//
		expect( wrapper ).toMatchSnapshot( );
		expect( findHeader3.length ).toBe( 1 );
		expect( findHeader3.text() ).toBe( category );
		expect( findLoading.exists() ).toBe( true );
		expect( findCard.length ).toBe( 1 );
		expect( findGridItem.exists() ).toBe( false );
	} );
	//
	test( 'useFetchGifs', ( ) => {
		//
		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false
		});
		//
		const wrapper		= shallow( <GiftGrid category={ category } /> );
		const findHeader3	= wrapper.find( 'h3' );
		const findLoading	= wrapper.find( '.loading' );
		const findCard		= wrapper.find( '.card-grid' );
		const findGridItem	= wrapper.find( 'GiftGridItem' );
		//
		expect( wrapper ).toMatchSnapshot( );
		expect( findHeader3.length ).toBe( 1 );
		expect( findHeader3.text() ).toBe( category );
		expect( findLoading.exists() ).toBe( false );
		expect( findCard.length ).toBe( 1 );
		expect( findGridItem.length ).toBe( gifs.length );

	} );
} );
