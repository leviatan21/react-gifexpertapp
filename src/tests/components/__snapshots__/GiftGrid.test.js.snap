// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en <GiftGrid /> Componente GiftGrid 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="loading animate__animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Puebas en <GiftGrid /> useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GiftGridItem
      id="3ohc1292yKn6Z1saGs"
      key="3ohc1292yKn6Z1saGs"
      title="getpunches GIF by mannyjammy"
      url="https://media3.giphy.com/media/3ohc1292yKn6Z1saGs/giphy.gif?cid=b1bbf17c1qvapj1ew4s6nuveegu7o50xrwvlagvr77c1reso&rid=giphy.gif"
    />
    <GiftGridItem
      id="l41m1H2BYnLRV26QM"
      key="l41m1H2BYnLRV26QM"
      title="Bruce Lee One Inch Punch GIF by Supercompressor"
      url="https://media2.giphy.com/media/l41m1H2BYnLRV26QM/giphy.gif?cid=b1bbf17c1qvapj1ew4s6nuveegu7o50xrwvlagvr77c1reso&rid=giphy.gif"
    />
  </div>
</Fragment>
`;
